### PR TITLE
Add block transforms preview

### LINKED ISSUE #26399: I should be able to preview block transforms before applying them
**Is your feature request related to a problem? Please describe.**
The only way to know what a block transform does to your block is to apply it, then undo the transformation if it doesn't work out. This is not an idea experience, and what could have been delightful has instead turned into a frustration moment. 

**Describe the solution you'd like**
A preview of the transform is available by hovering over each transform. This way I can see what my block will look like before I transform it. 

ScreenFlow

**Describe alternatives you've considered**
None.

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Resolves: https://github.com/WordPress/gutenberg/issues/26399

This PR adds previews to block transforms.

transformsPreview

### CODE REVIEW

File: packages/block-editor/src/components/block-switcher/block-styles-menu.js [hunk 1]
```diff
@@ -4,28 +4,49 @@
...
 	return (
 		<MenuGroup
 			label={ __( 'Styles' ) }
 			className="block-editor-block-switcher__styles__menugroup"
 		>
 			{ hoveredClassName && (
 				<PreviewBlockPopover
-					hoveredBlock={ hoveredBlock }
-					hoveredClassName={ hoveredClassName }
+					blocks={
```
Comment (gziolo) [inline packages/block-editor/src/components/block-switcher/block-styles-menu.js:33]: Block styles can be applied only when a single block is selected. It's only changing the CSS class name applied so it feels like `blockType.example` shouldn't be needed here. However, it looks like some sort of design decision from the past. Maybe to account for the case when a block has not enough or too much content 🤷🏻‍♂️

**Comment by gziolo** [comment]
The preview for styles feels a little bit like a duplication but I believe it existed before:

Screen Shot 2020-12-23 at 08 43 40

For block transforms it looks great, both for a single block and multiple selected blocks 👏🏻

File: packages/block-editor/src/components/block-switcher/block-styles-menu.js (hunk 1 above)
Comment (ntsekouras) [reply packages/block-editor/src/components/block-switcher/block-styles-menu.js:33]: Yeah, I didn't look into it deeper for now, but it looks like it's been like this for the reason you mentioned (existing content..)

**Comment by ntsekouras** [comment]
> Did it always work only when hovering an item?

Yes. I actually checked it myself when implementing because it made sense to show the preview with keyboard navigation. Maybe in a follow up? What do you think?

**Comment by gziolo** [comment]
Sure, a follow-up to align behavior is a great idea. I wasn't sure how it works in the main branch.

**Review by gziolo — APPROVED** [review]
Code wise everything is good. It would be nice to get another feedback oriented towards behavior.

**Comment by jasmussen** [comment]
Thanks for the gif. From a quick glance at it, it seems like a cool feature!

Except perhaps for the style variation transform, those are already thumbnails, it feels a little duplicate. It's fine probably? And maybe we can revisit?

**Comment by youknowriad** [comment]
Same it makes me wonder whether we should just show a simple list for the style variations too.

**Comment by ntsekouras** [comment]
> Except perhaps for the style variation transform, those are already thumbnails, it feels a little duplicate. It's fine probably? And maybe we can revisit?

This had been also discussed in my first GB PR: https://github.com/WordPress/gutenberg/pull/23028.
I can create a new issue for revisiting the change of styles display, as it will definitely take more time to discuss the design.

**Comment by mtias** [comment]
This is great — nice that it works with multi-transforms as well.

**Pull request merged**